### PR TITLE
Pin gdal to 3.3.x

### PR DESCRIPTION
**Pull request by weiji14**
Bump `gdal` from 3.2.x to 3.3.x. Supersedes #347. This is inline with conda-forge's current migration to gdal=3.3.x (see https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/1506 and https://conda-forge.org/status/#gdal33).

Update on rapidsai/integration is at https://github.com/rapidsai/integration/pull/300.

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by ajschmidt8** [comment]
ok to test

**Comment by weiji14** [comment]
There are 2 failing tests. Not quite sure what to do with these though :sweat_smile:

```bash
[  FAILED  ] PIPRefineTestLarge/0.TestLarge, where TypeParam = float (1063 ms)
[----------] 1 test from PIPRefineTestLarge/0 (1063 ms total)

[----------] 1 test from PIPRefineTestLarge/1, where TypeParam = double
[ RUN      ] PIPRefineTestLarge/1.TestLarge
../tests/utilities/column_utilities.cu:68: Failure
Expected equality of these values:
  lhs.size()
    Which is: 0
  rhs.size()
    Which is: 885
../tests/utilities/column_utilities.cu:68: Failure
Expected equality of these values:
  lhs.size()
    Which is: 0
  rhs.size()
    Which is: 885
../tests/utilities/column_utilities.cu:68: Failure
Expected equality of these values:
  lhs.size()
    Which is: 0
  rhs.size()
    Which is: 885
[  FAILED  ] PIPRefineTestLarge/1.TestLarge, where TypeParam = double (30 ms)
[----------] 1 test from PIPRefineTestLarge/1 (30 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 2 test suites ran. (1093 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] PIPRefineTestLarge/0.TestLarge, where TypeParam = float
[  FAILED  ] PIPRefineTestLarge/1.TestLarge, where TypeParam = double

 2 FAILED TESTS
```

**Comment by ajschmidt8** [comment]
> There are 2 failing tests. Not quite sure what to do with these though 

@thomcom, any thoughts here?

**Comment by thomcom** [comment]
For some reason I missed this in my github notifications feed, I'll take a look at it this week. Thanks!

**Comment by harrism** [comment]
@thomcom any update? Moving to 21.10, please retarget the branch.

**Comment by thomcom** [comment]
rerun tests

File: ci/gpu/build.sh [hunk 1]
```diff
@@ -46,7 +46,7 @@ gpuci_mamba_retry install "cudf=${MINOR_VERSION}.*" "cudatoolkit=$CUDA_REL" \
 
 # https://docs.rapids.ai/maintainers/depmgmt/ 
 # gpuci_mamba_retry remove --force rapids-build-env
-# gpuci_mamba_retry install "your-pkg=1.0.0"
+# gpuci_mamba_retry install "gdal=3.3.1"
```
Comment (weiji14) [inline ci/gpu/build.sh:49]:
```suggestion
# gpuci_mamba_retry install "your-pkg=1.0.0"
```

**Review by weiji14 — COMMENTED** [review]
Did the tests pass? And can we move on with https://github.com/rapidsai/integration/pull/300?

**Comment by Ethyling** [comment]
rerun tests

**Review by cwharris — APPROVED** [review]

**Review by cwharris — CHANGES_REQUESTED** [review]
errr, didn't realize this was wip.

**Comment by thomcom** [comment]
rerun tests

**Comment by Ethyling** [comment]
rerun tests

**Comment by thomcom** [comment]
@cwharris any chance you can reformat this pr with `scripts/run-clang-format.py` assuming you already have clang-format 11? Mine is 8.0.1 in conda and I'm not confident of how to get to 11.0.0 quickly.

**Comment by thomcom** [comment]
@gpucibot merge

**Review by cwharris — APPROVED** [review]

**Review by thomcom — APPROVED** [review]

**Comment by thomcom** [comment]
@gpucibot merge

**Comment by thomcom** [comment]
@dillon-cullinan can I get an approve 420?

**Review by ajschmidt8 — APPROVED** [review]

**Comment by thomcom** [comment]
@gpucibot merge

**Pull request merged**